README.md

```diff
@@ -81,72 +81,57 @@ but as necessary, rebuild the image using Step 1.
+13. Enter a name in the "Key Name" field, and under "Redirect URIs" enter:
+    `https://{ccm_app_hostname}/canvas/returnFromOAuth`
+    Where `{ccm_app_hostname}` is the ngrok hostname copied earlier (in step 2).
+    ***Note:*** Do ***NOT*** use the "Redirect URI (Legacy)" field.
+14. Enable the "Enforce Scopes" option, then add all scopes needed by the application (i.e., those listed in `ccm_web/server/src/canvas/canvas.scopes.ts`).
+15. Click the "Save" button.
+16. Copy the ID number of the API key created in Canvas. The ID is the long number shown in the "Details" column of the "Developer Keys" page. It usually looks like "`17700000000000nnn`".
+17. Click the "Show Key" button underneath the ID located in step 15, and copy the secret that appears in the dialog.
+18. Click the "ON" part of the switch in the "State" column of your API key, so that it has a green background.
+19. Make a `.env` file for the project, based on the sample provided.
+    `cp config/.env.sample .env`
+20. Edit the `.env` file.  The keys in the following list must/should be updated.
+
+    1. `DOMAIN` – Hostname of the server hosting the CCM application.
+       For local development purposes with ngrok, use the hostname
+       copied during step 2.
+    2. Verify the `LTI_PLATFORM_URL` variable value is correct for the instance of Canvas used.
+    3. Add to the `LTI_CLIENT_ID` variable value the LTI key ID number copied earlier (in step 10).
+    4. `CANVAS_INSTANCE_URL` – For U-M development/testing, this should
+       be `https://canvas-test.it.umich.edu`.  Note that this is the URL
+       that would be used by the user to access Canvas.  It may be 
+       different from the value of the `LTI_PLATFORM_URL` key.
+    5. Add to the `CANVAS_API_CLIENT_ID` variable value the API key copied earlier (in step 16).
+    6. Add to the `CANVAS_API_SECRET` variable value the secret copied earlier (in step 17).
+21. Build and start the application with docker-compose.
 
     1. `docker-compose build`
-    2. *Optional:* The very first time the database container starts, it takes some time to set itself up, during which it will not accept connections from the application container.  The application container then fails, which prevents it from creating tables in the DB it will need later.  A solution that seems to work is to start the database container alone first, letting it run in detached mode for a few minutes before starting the application.
-       `docker-compose up -d -- database`
-    3. `docker-compose up`
-
-23. Add the app to a course.  Go to (https://umich.instructure.com/courses/nnnnnn/settings/configurations), where "nnnnnn" is the course ID.
+    2. `docker-compose up`
+22. Add the app to a course.  Go to 
+    (https://canvas-test.it.umich.edu/courses/nnnnnn/settings/configurations), where "nnnnnn" is the course ID.
```

Comment:
I'm not sure it makes sense to include a link here, because of the different environments. I think I'd just say go to your course in Canvas, then click "Settings", as the first step here.

Reply:
I'll add environment-agnostic instructions.